### DISCUSSION TITLE
CD music support?

### DISCUSSION THREAD

**Discussion by yan487m** [ihhub/fheroes2#3988] [Q&A]
Hi,

Is CD music currently supported?

I got the original CDROM, in the settings there is an external music option, but it's always the MIDI played.
I tried to copy the files (.wav) on the installation directory (root), or to make a music folder, with no results.

Thanks a lot for your amazing work.

**Comment by oleg-derevenetz** [comment]
@yan487m External music is supported in OGG format. You will need to create folder with name `MUSIC` either in installation directory or in the user's data directory and place OGG files with appropriate names here. You can get the necessary file names for tracks from [this script](https://github.com/ihhub/fheroes2/blob/master/src/tools/cdinfo.py).

**Reply by yan487m** [reply]
Thx a lot again, these musics are so wonderful, i still remember all of them after all these years.

**Comment by yan487m** [comment]
I noticed that the volume is automatically lowered sometimes, without any reason.
In the Windows 10 volume mixer, fheroes2 application volume goes down, and don't go back to normal automatically.

Tested on 2 different computers.

**Reply by oleg-derevenetz** [reply]
There is already an existing issue with similar symptoms - #2009. However, this effect is hard to reproduce and also it seems that it's a Windows-only, because I never encountered it on my macOS devices (as well as on Windows BTW, but I almost don't use Windows).

**Reply by Kiiro-Yakumo** [reply]
I would argue that it may be Windows 10 attempt to be "smart" and it lowers the volume of the fheroes2 when it detects - or at least is convinced of - some other sound, like an alert so you can hear the alert even if the game will be "loud".

I don't remember at the moment where it is as I don't have Windows in home but... You may check something like - Googling images will get me only so far it seems - Control Panel -> Sound -> Communications -> [x] Do Nothing

If it will persist then you would have to get some kind of sound mixer visible while playing the game and observe it.

Oh also, I didn't notice such a behavior on GNU/Linux or PS Vita so it does sound like some Windows exclusive thing...
